Always show debug information in the info output

calavera opened the pull request:
Add also regression tests for #19893.

Signed-off-by: David Calavera david.calavera@gmail.com

calavera committed:
Add regression tests for client debug flag.

- Add client debug info to the `docker info` command.

Signed-off-by: David Calavera <david.calavera@gmail.com>

duglin commented on docker/client_test.go:
```diff
@@ -0,0 +1,23 @@
+	if logrus.GetLevel() != logrus.DebugLevel {
+		t.Fatalf("expected logrus debug level, got %v", logrus.GetLevel())
+	}
+}
```
Can you add an integration test to verify that the debug flag makes it all the way into the client code itself? e.g. do the equivalent of `docker -D run busybox echo hi` and make sure we see the DEBUG output to the screen?

cpuguy83 replied:
I'm not sure this is a worthwhile venture since this debug messages are not something we should necessarily even see and could easily be changed around.

duglin replied:
I wasn't looking to diff the exact output from the cli debug msgs, rather just something with the word "[DEBU]" it in or to make sure "docker -D info" shows debug info ( ala https://github.com/docker/docker/issues/16264#issuecomment-178250407 ). Just something to prove it get past the flag parsing logic and into the rest of the code.

cpuguy83 commented:
LGTM

calavera commented:
I added a test to check `docker info`'s output.

cpuguy83 commented on integration-cli/docker_cli_info_test.go:
```diff
@@ -146,3 +146,21 @@ func (s *DockerSuite) TestInfoDisplaysStoppedContainers(c *check.C) {
 	c.Assert(out, checker.Contains, fmt.Sprintf(" Stopped: %d\n", 1))
 }
+
+func (s *DockerSuite) TestInfoDebug(c *check.C) {
```
The `--debug` flag on the client does not affect the output of `info` here.

duglin replied:
yup - I think you may need to do something like `docker -D run busybox echo hi` and then look for at least one "DEBU" in the output - if there, then it passes.

duglin replied:
Although, if we don't already have a daemon-side test verifying the `--debug` shows up in the "info" output then keeping this test for that would be good too  :-)

calavera commented:
@cpuguy83, @duglin your comments made me wonder why we don't show the debug info in the client at all times. It's also inconsistent that we show information about the server but not the client, I added both information to the `docker info` command.

I've also added tests to make sure the log is in debug mode in the server and that the logic to enable and disable debug mode is correct.

duglin commented:
@calavera makes sense

icecrime commented:
LGTM
